### PR TITLE
Add new integration for Parad

### PR DESCRIPTION
**Pull request by LICHTENBERGERHugo**

### CODE REVIEW

File: Daspren/daspren-parad/tests/file_opened.json
Comment (squioc) [inline Daspren/daspren-parad/tests/file_opened.json:1]: Can you apply the linter on this file?
In the directory `utils`, type `poetry utils/linter.py fix --changes`

File: Daspren/daspren-parad/tests/malicious_behavior.json
Comment (squioc) [inline Daspren/daspren-parad/tests/malicious_behavior.json:1]: Can you apply the linter on this file?
In the directory `utils`, type `poetry utils/linter.py fix --changes`

File: Daspren/daspren-parad/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  - name: set_fields
+
+stages:
+  set_recommended:
+    actions:
+      - set:
+        event.kind: "event"
+        observer.vendor: "Daspren"
+        observer.product: "Parad"
+        observer.type: "dlp-solution"
```
Comment (squioc) [inline Daspren/daspren-parad/ingest/parser.yml:19]: Indentation issue.
```suggestion
          observer.vendor: "Daspren"
          observer.product: "Parad"
          observer.type: "dlp-solution"
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for your contribution.
There are some changes to apply to the format. Thank you.

File: Daspren/daspren-parad/tests/file_opened.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "sekoiaio": {
+      "intake": {
+        "dialect": "Daspren Parad",
+        "dialect_uuid": "23813540-b658-48dd-b030-e9b92168bbf4"
+      }
+    }
+  },
+  "expected": {
+    "message": "{\"time\":\"2024-03-04T10:43:31.337737900+01:00\",\"level\":\"INFO\",\"message\":\"File opened\",\"event_data\":{\"basename\":\"AppData\",\"created_length\":\"0\",\"executable\":\"C:\\Windows\\System32\\RuntimeBroker.exe\",\"executable_basename\":\"RuntimeBroker.exe\",\"executable_category\":\"System\",\"fullpath\":\"C:\\Users\\Testeur\\AppData\",\"fullpath_category\":\"User\",\"hash\":\"6b1070d18921913ef663562e238873717fff4f2afe3e7672c226e91adc614dea\",\"pid\":\"4960\",\"ppid\":\"0\",\"signed\":\"true\"},\"channel\":\"remote\",\"env_id\":\"7ba0a633-f8a3-410b-ba6f-5974705ced3a\",\"parad_version\":\"0.6.1\",\"os\":\"Windows 10 Pro\",\"machine_name\":\"bloquant\",\"msg_id\":6094}",
```
Comment (squioc) [inline Daspren/daspren-parad/tests/file_opened.json:12]: We detected that the json in `message` seems to be malformed. Can you check it? Thanks.
According to our analysis, it seems to be the backslashes in the fields `executable` and `fullpath` that generate the issue.

**Review by squioc — CHANGES_REQUESTED** [review]

File: Daspren/daspren-parad/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        input_field: "{{original.message}}"
+        output_field: event
+  - name: set_recommended
+  - name: set_fields
+
+stages:
+  set_recommended:
+    actions:
+      - set:
+        event.kind: "event"
```
[the remark this answers is not in the document]
Comment (LICHTENBERGERHugo) [reply Daspren/daspren-parad/ingest/parser.yml]: Done

File: Daspren/daspren-parad/ingest/parser.yml (hunk 1 above)
Comment (LICHTENBERGERHugo) [reply Daspren/daspren-parad/ingest/parser.yml]: Done

File: Daspren/daspren-parad/tests/file_opened.json
Comment (squioc) [inline Daspren/daspren-parad/tests/file_opened.json:1]: can you update the test according to the last changes? Thanks
```
$ cd utils/
$ poetry run pytest test_formats.py --format=daspren-parad --fix-expectations
```

File: Daspren/daspren-parad/tests/malicious_behavior.json
Comment (squioc) [inline Daspren/daspren-parad/tests/malicious_behavior.json:1]: Can you update the test according to the last changes? Thanks
```
$ cd utils/
$ poetry run pytest test_formats.py --format=daspren-parad --fix-expectations
```

File: Daspren/daspren-parad/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          observer.type: "dlp-solution"
+  set_fields:
+    actions:
+      - set:
+          "@timestamp": "{{json.event.time}}"
+          process.name: "{{json.event.executable_basename}}"
+          process.executable: "{{json.event.executable}}"
+          process.pid: "{{json.event.pid}}"
+          process.parent.pid: "{{json.event.ppid}}"
+          daspren.process.hash.blake3: "{{json.event.hash}}"
```
Comment (squioc) [inline Daspren/daspren-parad/ingest/parser.yml:27]: According to the tests, file_opened.json's event contains information in the `event_data` property, while for the malicious_behaviour.json's message contains information at the root.
We suggest modifying these lines:
```suggestion
          process.name: "{{json.event.executable_basename or json.event.event_data.executable_basename}}"
          process.executable: "{{json.event.executable or json.event.event_data.executable}}"
          process.pid: "{{json.event.pid or json.event.event_data.pid}}"
          process.parent.pid: "{{json.event.ppid or json.event.event_data.ppid}}"
          daspren.process.hash.blake3: "{{json.event.hash or json.event.event_data.hash}}"
```

File: Daspren/daspren-parad/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          process.code_signature.exists: "{{json.event.signed}}"
+          host.hostname: "{{json.event.machine_name}}"
+          host.name: "{{json.event.machine_name}}"
+          host.os.type: "{{json.event.os}}"
+          agent.version: "{{json.event.parad_version}}"
+          agent.id: "{{json.event.env_id}}"
+          event.action: "{{json.event.message}}"
+          file.name: "{{json.event.basename}}"
+          file.path: "{{json.event.fullpath}}"
```
Comment (squioc) [inline Daspren/daspren-parad/ingest/parser.yml:36]: sames here:
```suggestion
          file.name: "{{json.event.basename or json.event.event_data.basename}}"
          file.path: "{{json.event.fullpath or json.event.event_data.fullpath}}"
```

File: Daspren/daspren-parad/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  set_fields:
+    actions:
+      - set:
+          "@timestamp": "{{json.event.time}}"
+          process.name: "{{json.event.executable_basename}}"
+          process.executable: "{{json.event.executable}}"
+          process.pid: "{{json.event.pid}}"
+          process.parent.pid: "{{json.event.ppid}}"
+          daspren.process.hash.blake3: "{{json.event.hash}}"
+          process.code_signature.exists: "{{json.event.signed}}"
```
Comment (squioc) [inline Daspren/daspren-parad/ingest/parser.yml:28]: same here:
```suggestion
          process.code_signature.exists: "{{json.event.signed or json.event.event_data.signed}}"
```

**Review by squioc — CHANGES_REQUESTED** [review]
Hello,
we detected some mismatching between the parser and the tests.
We add some suggestions

**Pull request merged**